### DISCUSSION TITLE
USB connector description in error?

### DISCUSSION THREAD

**Discussion by BitLoose** [df8oe/UHSDR#1941] [General]
In M0NKA's user manual it states that the USB DFU connector is for Device Firmware updates, but in the manual here it states the reverse, is there an error?

DF8OE mchf manual:
USB-A (HOST)
Used for loading Firmware updates into the mcHF. "USB-A" type.

mini-USB (DFU)
Used for Rig control (CAT: TS-817) and Digital audio in/out (48 kHz). "Mini-B" type.

**Comment by df8oe** [comment]
No, that's correct. M0NKA and UHSDR is not the same!

**Reply by BitLoose** [reply]
Thanks for letting me know, a bit unclear as M0NKA's UHSDR GitHub page states that "This branch is up to date with df8oe/UHSDR:active-devel."

Was the use of the USB ports swapped at some time, changing the PA11,12 USB port from firmware update usage to PC USB sound card, CAT etc usage?

**Reply by BitLoose** [reply]
I´m trying to get the USB "device" working for CAT, but plugging the micro-USB (PA11-12 USB device) into my laptop Windows doesn´t detect it.
Any ideas what could be the problem?

**Comment by df8oe** [comment]
The state that USB-A can handle USB stick for updates was about 5 years implemented. Before this USB-A was unused. DFU and cat/audio never have other connector than mini-USB.

Can you test using Linux? I do not have Windows and can not give any hint. But as I remeber Windows recognizes audio and serial port for cat out-of-the-box (without installation of drivers) as Linux does.

**Reply by BitLoose** [reply]
Is it the USB device using PA10 an 11 that are for the audio and CAT interface?
I'm checking the connections to see if there is a fault.

**Reply by db4ple** [reply]
I don't have the schematics at hand. But the Mini-USB is the connection which carries CAT and Audio. In any case, when you plug this into Windows, it must show new devices. Audio devices are "self-installing", the serial port requires installation of STM drivers (described in the documentation). Just open the Windows Device Manager and see if something happens when you plugin the the TRX. There should be new audio devices showing up. If not, USB cable or internal connection is broken/damaged.

**Reply by db4ple** [reply]
BTW, it would be super helpful if you would provide a clear description what device and firmware you are using.

**Reply by BitLoose** [reply]
I'm using the firmware from M0NKA's GitHub, which says it's up-to-date with the DF8OE branch.  I've been comparing the usbd_conf.c files and they seem identical.
Changed the microUSB to USB2 cable and Windows now detects the microUSB cable but is reporting unknown device, invalid configuration descriptor, so some progress.
Plugging in the cable, all the USB interrupts are firing, first is HAL_PCD_SuspendCallback, but connect doesn't get called.

**Reply by BitLoose** [reply]
This is the error: "Windows has stopped this device because it has reported problems. (Code 43)
The USB device returned an invalid USB configuration descriptor."
On the web people say it can be a driver problem, or hardware, but haven't found anything specific yet.

**Reply by BitLoose** [reply]
Just found that the USB Audio middleware files are very different, and the USB descriptor too, so M0NKA´s GitHub isn't the same after all.
This could be the problem, I'll see if I can patch in the newer descriptors and stuff.  All part of the fun hey

**Reply by db4ple** [reply]
Well, we here can only support UHSDR software from this repo and are unfortunately not able to support the other firmware due to lack of knowledge.

**Reply by BitLoose** [reply]
Yes of course, so I changed over to your version and it's working now, at least a "USB Sound Card" appears in the device manager.
Trying to get it to work with HDSDR, I think it's listing as a microphone, but no audio data is arriving, probably got to set an option in the mcHF menu somewhere.

**Reply by BitLoose** [reply]
It's working with HDSDR on Windows. Quite a relief!! 73

**Reply by db4ple** [reply]
Great 👍
